src/mono/mono/mini/simd-intrinsics.c

```diff
@@ -2815,7 +2792,11 @@ emit_vector_2_3_4 (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignature *f
 	case SN_Negate:
 	case SN_op_UnaryNegation: {
 #if defined(TARGET_ARM64) || defined(TARGET_AMD64)
-		return emit_simd_ins (cfg, klass, OP_NEGATION, args [0]->dreg, -1);
+		if (!COMPILE_LLVM(cfg)) {
+			return emit_simd_ins_for_unary_op (cfg, klass, fsig, args, MONO_TYPE_R4, id);
```

Comment:
Why is this condition needed?

Reply:
For !LLVM, arg_type is not being set if used `emit_simd_ins`, causing this assertion to be reached : https://github.com/dotnet/runtime/blob/15e9370e32ab7a2c25d56e95554ff69d334bff62/src/mono/mono/mini/mini-arm64.c#L467

Reply:
Then, both of the LLVM and non-LLVM case could use `emit_simd_ins_for_unary_op`, right?

Reply:
Went with the second option.